client/modules/Manage/Bookings/BookingsAdm.js

```diff
@@ -8,24 +8,188 @@ import {
 
 export class BookingsAdm extends Component {
+  ALL_USERS = "All Users";
+  APPROVED_USERS = "Users with APPROVED licenses";
+  PENDING_USERS = "Users with PENDING licenses";
+
   constructor(props) {
     super(props);
+    this.state = {
+      bookings: [],
+      users: [],
+      cars: [],
+      offers: [],
+      selectedBooking: null,
+      savedAlertOpen: false,
+      errorAlertOpen: false,
+      filter: this.ALL_USERS,
+      dropdownsOpen: {
+        filter: false,
+      },
+    };
+  }
+
+  componentDidMount() {
+    this.getBookings();
+    this.getUsers();
+    this.getCars();
+  }
+
+  getBookings() {
+    manageSvc.bookings.getAll()
+      .then(res => {
+        let tmp = null;
+        if (this.state.selectedBooking) {
+          tmp = res.data.find(f => f._id === this.state.selectedBooking._id);          
+        }
+        this.setState({
+          bookings: res.data,
+          selectedBooking: tmp,
+        });
+      })
+      .catch(e => console.log(e));
+  }
+
+  getUsers() {
+    manageSvc.users.getAll()
+      .then(res => this.setState({users: res.data}))
+      .catch(e => console.log(e));
+  }
+
+  getCars() {
+    manageSvc.cars.getAll()
+      .then(res => this.setState({cars: res.data}))
+      .catch(e => console.log(e));
+  }
+
+  getOffers() {
+    manageSvc.offers.getAll()
+      .then(res => this.setState({offers: res.data}))
+      .catch(e => console.log(e));
+  }
+
+  handleBookingSelected(evt) {
+    // evt is an array of objects
+    this.setState({selectedBooking: evt[0]});
+  }
+
+  handleSaved(savedOffer) {
+    this.getBookings();
+    //Dismiss any previous error
+    this.dismissError();
+    this.setState({ savedAlertOpen: true });
+  }
+
+  handleError() {
+    this.getBookings();
+    //Dismiss any previous success
+    this.dismissSuccess();
+    this.clearSearch();
+    this.setState({ errorAlertOpen: true });
+  }
+
+  selectedBookingForm() {
+    return (
+      !this.state.selectedBooking 
+      ? '' 
+      : <BookingAdm
+          onSaved={this.handleSaved.bind(this)}
+          onError={this.handleError.bind(this)}
+          booking={this.state.selectedBooking}
+          cars={this.state.cars}
+          users={this.state.users}
+          offers={this.state.offers}/>
+    );
+  }
+
+  toggle(e) {
+    const val = !this.state.dropdownsOpen.filter;
+    this.setState({
+      dropdownsOpen: { filter: val },
+    })
+  }
+
+  clearSearch() {
+    this.refs.typeahead.getInstance().clear();
+    this.setState({selectedBooking: null});
+  }
+
+  dismissSuccess() {
+    this.setState({ savedAlertOpen: false });
+  }
+
+  dismissError() {
+    this.setState({ errorAlertOpen: false });
   }
 
   render() {
+
     return (
       <div className={stylesMain.body}>
-        <div className="row">
+        <Row>
           <Col>
             <h1 className={stylesMain.title}>Manage Bookings</h1>
           </Col>
-        </div>
+        </Row>
+        
+        <Row>
```

Comment:
no need for empty row

Reply:
Oh thanks... Was attempting something else there. Forgot to remove entirely